django_cloud_deploy/tests/e2e/gae_deploy_test.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Versconsolen 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITconsoleNS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissconsolens and
+# limitatconsolens under the License.
+"""End to end test for create and deploy new project."""
+
+import os
+import shutil
+import tempfile
+import types
+import unittest
+import urllib.parse
+
+import backoff
+from django_cloud_deploy.cli import io
+from django_cloud_deploy.cli import new
+from django_cloud_deploy.cli import update
+from django_cloud_deploy.tests.lib import test_base
+from django_cloud_deploy.tests.lib import utils
+import requests
+from selenium import webdriver
+from selenium.webdriver.chrome import options
+
+
+class GAEDeployAndUpdateE2ETest(test_base.ResourceCleanUpTest):
+    """End to end test for create and deploy new project."""
+
+    _CLOUDSQL_ROLES = ('roles/cloudsql.client', 'roles/cloudsql.editor',
+                       'roles/cloudsql.admin')
+
+    _FAKE_CLOUDSQL_SERVICE_ACCOUNT = {
+        'id': utils.get_resource_name('sa'),
+        'name': 'Fake CloudSQL Credentials',
+        'file_name': 'credentials.json',
+        'roles': _CLOUDSQL_ROLES
+    }
+
+    def setUp(self):
+        super().setUp()
+        self.project_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.project_dir)
+
+    @staticmethod
+    @backoff.on_exception(
+        backoff.expo, requests.exceptions.ConnectionError, max_tries=3)
+    def _get_with_retry(url: str) -> requests.models.Response:
+        return requests.get(url)
+
+    @unittest.mock.patch('portpicker.pick_unused_port', return_value=5432)
+    def test_deploy_and_update_new_project(self, unused_mock):
+        # Generate unique resource names
+        fake_superuser_name = 'admin'
+        fake_password = 'fake_password'
+
+        cloud_storage_bucket_name = utils.get_resource_name('bucket')
+        django_project_name = utils.get_resource_name('project', delimiter='')
+        service_name = utils.get_resource_name('svc', delimiter='')
+
+        # Generate names we hardcode for users
+        service_account_email = '{}@{}.iam.gserviceaccount.com'.format(
+            self._FAKE_CLOUDSQL_SERVICE_ACCOUNT['id'], self.project_id)
+        member = 'serviceAccount:{}'.format(service_account_email)
+
+        with self.clean_up_bucket(cloud_storage_bucket_name), \
```

Comment:
Maybe in a follow-up PR it would be possible to move this prompt stuff into a common file?

Reply:
Will do this in another PR.